EIPS/eip-541.md

```diff
@@ -0,0 +1,139 @@
+# Preamble
+
+    EIP: <to be assigned>
+    Title: Mineable Token Standard
+    Author: Jay Logelin <jlogelin@fas.harvard.edu>
+    Type: Standard
+    Category: ERC
+    Status: Draft
+    Created: 2018-03-07
+    Requires: ERC-20
+
+# Simple Summary
+A standard interface for mineable tokens.
+
+# Abstract
+The following standard allows for the implementation of a standard API for mineable tokens within smart contracts. This standard provides basic functionality to mine and reward ERC20 tokens using a Proof of Work scheme.
+
+# Motivation
+A standard interface allows any Mineable ERC20 Tokens on Ethereum to be handled by general-purpose applications. Tokens within the Ethereum ecosystem may still require economic incentivization that proof of work mining provides. Specifically, the smart contract implementation will allow for Mineable ERC20 tokens to be tracked in standardized wallets, traded on exchanges, and interfaced with standardized third party mining software and equipment.
+
+# Specification
+## Token
+### ERC-20 compatibility Methods
+
+#### name
+
+Returns the name of the token - e.g. `"0xBitcoin Token"`.
+
+OPTIONAL - This method can be used to improve usability,
+but interfaces and other contracts MUST NOT expect these values to be present.
+
+``` js
+function name() constant returns (string name)
+```
+
+#### symbol
+
+Returns the symbol of the token. e.g. `"0xBTC"`.
+
+OPTIONAL - This method can be used to improve usability,
+but interfaces and other contracts MUST NOT expect these values to be present.
+
+``` js
+function symbol() constant returns (string symbol)
+```
+
+#### totalSupply
+
+Returns the total token supply.
+
+``` js
+function totalSupply() constant returns (uint256 totalSupply)
+```
+
+#### balanceOf
+
+Returns the account balance of another account with address `_owner`.
+
+``` js
+function balanceOf(address _owner) constant returns (uint256 balance)
+```
+
+### Mining Operations
+
+
+#### mint
+
+Returns a flag indicating a successful hash digest verification. In order to prevent MiTM attacks, it is recommended that the digest include a recent ethereum block hash and msg.sender's address. Once verified, the mint function calculates and delivers a mining reward to the sender and performs internal accounting operations on the contract's supply.
+
+``` js
+function mint(uint256 nonce, bytes32 challenge_digest) public returns (bool success)
```

Comment:
*Suggestion*: Allow merge mining by including a new parameter `uint256 extraNonce`
*Reasoning*: Merge-mining works by concatenating nonce + extraNonce and then performing PoW on the concatenation. Let's assume extra-nonce is just a piggy-back token. Whenever a modified miner is able to successfully mine a block with the extra-nonce information in it, receives both the original token, as well as the piggyback token. 

*Example*: 0xNamecoin implemented on top of 0xBitcoin. 0xNamecoin information (dns mapping) is hashed and the merkle root is then used as the extraNonce field. If a user running a 0xNamecoin miner successfully runs a mint operation, it recieves 0xNamecoin tokens.

*Resources*: 
https://bitcointalk.org/index.php?topic=1040859.0
https://namecoin.org/docs/faq/
https://github.com/bitcoin/bitcoin/blob/44080a90a29292df96e92f22242785c5040000a1/src/miner.h#L196
http://mmpool.org/main